code/datums/callback.dm

```diff
@@ -100,3 +100,60 @@
 	if (object == GLOBAL_PROC)
 		return call(delegate)(arglist(calling_arguments))
 	return call(object, delegate)(arglist(calling_arguments))
+
+
+/datum/callback_select
+	var/list/finished
+	var/pendingcount
+	var/total
+
+/datum/callback_select/New(count, savereturns)
+	total = count
+	if (savereturns)
+		finished = new(count)
+
+
+/datum/callback_select/proc/invoke_callback(index, datum/callback/callback, list/callback_args, savereturn = TRUE)
+	set waitfor = FALSE
+	if (!callback || !istype(callback))
+		//This check only exists because the alternative is callback_select would block forever if given invalid data
+		CRASH("invalid callback passed to invoke_callback")
+	if (!length(callback_args))
+		callback_args = list()
+	pendingcount++
+	debug_usr("calling callback")
+	var/rtn = callback.Invoke(arglist(callback_args))
+	debug_usr("callback returned")
+	pendingcount--
+	if (savereturn)
+		finished[index] = rtn
+
+
+
+
+//runs a list of callbacks asynchronously, returning once all of them return.
+//callbacks can be repeated.
+//callbacks-args is a optional list of argument lists, in the same order as the callbacks,
```

Comment:
the class or is there something on this line?

if I'm getting this right a callback_select is a way to call different callbacks in a group OR the same one multiple times (or a mix of both)? if so I'd prefer something like callback_collection, rather than the SQL-like "select" name.